### PR TITLE
egl: add __ENABLE_EGL__ to on/off egl compilation

### PR DESCRIPTION
**Pull request by LindaYu17**
Signed-off-by: Linda Yu linda.yu@intel.com

### CODE REVIEW

File: configure.ac
Comment (chivakker) [inline configure.ac:118]: What is that you want to accomplish here?  How is your environment failing without this?

File: egl/egl_util.c
Comment (chivakker) [inline egl/egl_util.c]: In principle, we should not have a ENABLE_EGL within a EGL file.  If we're compiling this on non-egl scenarios that has to be fixed.

File: egl/egl_util.h
Comment (chivakker) [inline egl/egl_util.h]: same as in egl_util.c file comment.

File: egl/egl_vaapi_image.cpp
Comment (chivakker) [inline egl/egl_vaapi_image.cpp]: same as in egl_util.c file comment

File: egl/egl_vaapi_image.h
Comment (chivakker) [inline egl/egl_vaapi_image.h]: same as in egl_util.c file comment

File: v4l2/v4l2_codecbase.h
Comment (chivakker) [inline v4l2/v4l2_codecbase.h:108]: This virtual method can be defined regardless of configure flag enable_egl.  Then the implementations should consider adding the enable_egl if required.

**Review by chivakker — COMMENTED** [review]
Can you please elaborate on the problem that you want to resolve with this patch? Thanks.

**Comment by xuguangxin** [comment]
Hi @LindaYu17 , is any update for Daniel's comments

File: configure.ac
Comment (LindaYu17) [reply configure.ac:118]: just want to use ENABLE_EGL to simply disable/enable it because in many scenarios we don't need to compile egl related code to the library.

File: egl/egl_util.c
Comment (LindaYu17) [reply egl/egl_util.c]: thanks for the comments.
updated.
decide to compile it or not in v4l2/makefile.am

File: v4l2/v4l2_codecbase.h
Comment (chivakker) [reply v4l2/v4l2_codecbase.h:108]: Nevermind the above comment.  EGLDisplay, EGLContext are only defined with ENABLE_EGL so this is ok

File: configure.ac
Comment (chivakker) [reply configure.ac:118]: Ok, it makes sense, thanks for the explanation.

**Review by chivakker — APPROVED** [review]
The change looks good to me now.  Please have someone verify it independently before merging.

**Comment by xuguangxin** [comment]
Hi Dongping:
Could help to verify, it should be buildable when we enable or disable egl. 
thanks

**Comment by wudping** [comment]
verify + 1

**Comment by xuguangxin** [comment]
seems related to https://github.com/01org/libyami-utils/pull/47, will merge this after https://github.com/01org/libyami-utils/pull/47 fixed

**Comment by xuguangxin** [comment]
@LindaYu17 , this will conflict with apache branch, please help to rebase it.
thanks

**Comment by LindaYu17** [comment]
@xuguangxin rebase done.

**Comment by xuguangxin** [comment]
merged as 18864d6a9b8748b8ae64abb32ad0a80886ecd11b thanks

**Pull request closed without merging**
